File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayBoolValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayBoolInValue.OptArrayBoolValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayBoolValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayBoolValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                boolean[].class,
                OptArrayBoolValue.class.getDeclaredField("optArrayBool").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                boolean[].class,
                OptArrayBoolValue.class.getMethod("getOptArrayBool").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayBoolValue.class,
                OptArrayBoolValue.class.getMethod("setOptArrayBool", boolean[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayBoolValue().getOptArrayBool());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayBoolValue.scala
package com.dslplatform.ocd.values

import OptArrayBoolInValue.OptArrayBoolValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayBoolValue extends SpecExtensions {
  def `Test property field type: Option[Array[Boolean]]`() =
    checkType(
      typeOf[Option[Array[Boolean]]]
    , typeOf[OptArrayBoolValue].member("optArrayBool": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayBoolValue().optArrayBool === None)
  }
}
